cmd/opentelemetry/app/exporter/elasticsearchexporter/spanstore.go

```diff
@@ -0,0 +1,209 @@
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry/app/exporter/elasticsearchexporter/esclient"
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry/app/exporter/elasticsearchexporter/estranslator"
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/cache"
+	"github.com/jaegertracing/jaeger/pkg/es/config"
+	"github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/dbmodel"
+)
+
+// EsSpanWriter holds components required for ES span writer
+type EsSpanWriter struct {
+	logger           *zap.Logger
+	client           esclient.ElasticsearchClient
+	serviceCache     cache.Cache
+	spanIndexName    indexNameProvider
+	serviceIndexName indexNameProvider
+	translator       *estranslator.Translator
+}
+
+// NewEsSpanWriter creates new instance ofr EsSpanWriter
+func NewEsSpanWriter(params config.Configuration, logger *zap.Logger) (*EsSpanWriter, error) {
+	client, err := esclient.GetElasticsearchClient(params, logger)
+	if err != nil {
+		return nil, err
+	}
+	tagsKeysAsFields, err := config.LoadTagsFromFile(params.Tags.File)
+	if err != nil {
+		return nil, err
+	}
+	return &EsSpanWriter{
+		client:           client,
+		spanIndexName:    newIndexNameProvider("jaeger-span", params.IndexPrefix, params.UseReadWriteAliases),
+		serviceIndexName: newIndexNameProvider("jaeger-service", params.IndexPrefix, params.UseReadWriteAliases),
+		translator:       estranslator.NewTranslator(params.Tags.AllAsFields, tagsKeysAsFields, params.GetTagDotReplacement()),
+		serviceCache: cache.NewLRUWithOptions(
+			100000,
+			&cache.Options{
+				TTL: time.Hour * 12,
+			},
+		),
+	}, nil
+}
+
+func newIndexNameProvider(index, prefix string, useAliases bool) indexNameProvider {
+	if prefix != "" {
+		prefix = prefix + "-"
+		index = prefix + index
+	}
+	index = index + "-"
+	if useAliases {
+		index = index + "write"
+	}
+	return indexNameProvider{
+		index:    index,
+		useAlias: useAliases,
+	}
+}
+
+type indexNameProvider struct {
+	index    string
+	useAlias bool
+}
+
+func (n indexNameProvider) get(date time.Time) string {
+	if n.useAlias {
+		return n.index
+	}
+	spanDate := date.UTC().Format("2006-01-02")
+	return n.index + spanDate
+}
+
+// CreateTemplates creates index templates.
+func (w *EsSpanWriter) CreateTemplates(spanTemplate, serviceTemplate string) error {
+	err := w.client.PuTemplate("jaeger-span", strings.NewReader(spanTemplate))
+	if err != nil {
+		return err
+	}
+	err = w.client.PuTemplate("jaeger-service", strings.NewReader(serviceTemplate))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO remove
+var counter int64
+
+// WriteTraces writes traces to the storage
+func (w *EsSpanWriter) WriteTraces(_ context.Context, traces pdata.Traces) (int, error) {
+	atomic.AddInt64(&counter, int64(traces.SpanCount()))
+	fmt.Printf("counter: %d\n", counter)
```

Comment:
To remove?

Reply:
I will remove the prints once it's ready. Just ignore them for now